File: api/Models/Hub.cs
using System.Text.Json.Serialization;

namespace CroApp.Api.Models;

// Id is a generated GUID, and Status is the partition key - the dominant query is "list
// every approved Hub" (a handful of rows, read far more than written), which a /status
// partition keeps single-partition even once a future "Pending" status (user-proposed
// Hubs, not built yet) starts accumulating its own rows. Unlike Waypoint, a Hub has no
// single user-owner - CreatedByUserId just records who placed it, it isn't a partition
// key or an ownership check the way Waypoint.UserId is. Status is a plain string (not an
// enum) so a future proposal/approval flow doesn't require a schema migration; only
// "Approved" is ever produced by the admin-placement flow this pass.
public record Hub(
    [property: JsonPropertyName("id")] string Id,
    string Name,
    double Latitude,
    double Longitude,
    string Status,
    string CreatedByUserId,
    DateTimeOffset CreatedAt,
    string? Category = null,
    string? ProfilePictureUrl = null);

public static class HubStatus
{
    public const string Approved = "Approved";
}

File: app/lib/models/hub.dart
// App-curated public landmark, placed by an admin through the map's "Add Hub" flow -
// unlike Waypoint, a Hub has no single user-owner (createdByUserId just records who
// placed it, not an ownership check).
class Hub {
  final String id;
  final String name;
  final double latitude;
  final double longitude;
  final String status;
  final String createdByUserId;
  final String? category;
  final String? profilePictureUrl;

  Hub({
    required this.id,
    required this.name,
    required this.latitude,
    required this.longitude,
    required this.status,
    required this.createdByUserId,
    this.category,
    this.profilePictureUrl,
  });

  factory Hub.fromJson(Map<String, dynamic> json) => Hub(
        id: json['id'] as String,
        name: json['name'] as String,
        latitude: (json['latitude'] as num).toDouble(),
        longitude: (json['longitude'] as num).toDouble(),
        status: json['status'] as String,
        createdByUserId: json['createdByUserId'] as String,
        category: json['category'] as String?,
        profilePictureUrl: json['profilePictureUrl'] as String?,
      );
}
